api/src/main/java/com/epam/pipeline/manager/datastorage/providers/gcp/GSBucketStorageHelper.java

```diff
@@ -55,35 +71,192 @@ public String createGoogleStorage(final GSBucketStorage storage) {
     }
 
     public void deleteGoogleStorage(final String bucketName) {
-        final Storage client = getClient();
+        final Storage client = gcpClient.buildStorageClient(region);
         final Iterable<Blob> blobs = client.list(bucketName, Storage.BlobListOption.prefix(EMPTY_PREFIX)).iterateAll();
         blobs.forEach(blob -> blob.delete());
-        final boolean deleted = client.delete(bucketName);
+        deleteBucket(bucketName, client);
+    }
 
-        if (!deleted) {
-            throw new IllegalArgumentException(String.format("Failed to delete google data storage %s", bucketName));
+    public DataStorageListing listItems(final GSBucketStorage storage, final String path, final Boolean showVersion,
+                                        final Integer pageSize, final String marker) {
+        final List<AbstractDataStorageItem> items = new ArrayList<>();
+
+        String requestPath = Optional.ofNullable(path).orElse(EMPTY_PREFIX);
+        if (StringUtils.isNotBlank(requestPath)) {
+            requestPath = normalizeFolderPath(requestPath);
         }
+        final Storage client = gcpClient.buildStorageClient(region);
+        final String bucketName = storage.getPath();
+        checkBucketExists(bucketName, client);
```

Comment:
Do we need to check that bucket exists for each method? Maybe `StorageClient` will throw appropriate exception itself?